### PR TITLE
add isotonic calibrators

### PR DESCRIPTION
**Pull request by KnathanM**
## Description
This reimplements Isotonic calibrators from v1

### CODE REVIEW

File: chemprop/uncertainty/calibrator.py [hunk 1]
```diff
@@ -257,12 +287,14 @@ def fit(self, uncs: Tensor, targets: Tensor, mask: Tensor) -> Self:
         Parameters
         ----------
         uncs: Tensor
-            the predicted uncertainties (i.e., the predicted probabilities for each class) of the shape of ``n x t x c``, where ``n`` is the number of input
-            molecules/reactions, ``t`` is the number of tasks, and ``c`` is the number of classes.
+            the predicted uncertainties (i.e., the predicted probabilities for each class) of the
```
Comment (KnathanM) [inline chemprop/uncertainty/calibrator.py:290]: I also make these lines shorter to stay less than 100 charactes

File: tests/unit/uncertainty/test_calibrators.py [hunk 2]
```diff
@@ -102,3 +140,50 @@ def test_PlattCalibrator(
...
+    cal_uncs, cal_targets, cal_mask, test_uncs, cal_test_uncs
+):
+    """
+    Testing the IsotonicMulticlassCalibratorCalibrator
+    """
+    calibrator = IsotonicMulticlassCalibrator()
+    calibrator.fit(cal_uncs, cal_targets, cal_mask)
+    uncs = calibrator.apply(test_uncs)
+    torch.set_printoptions(precision=10)
+    print(uncs)
```
Comment (shihchengli) [inline tests/unit/uncertainty/test_calibrators.py:188]:
```suggestion
```

File: tests/unit/uncertainty/test_calibrators.py [hunk 3]
```diff
@@ -102,3 +140,50 @@ def test_PlattCalibrator(
...
+def test_IsotonicMulticlassCalibratorCalibrator(
+    cal_uncs, cal_targets, cal_mask, test_uncs, cal_test_uncs
+):
+    """
+    Testing the IsotonicMulticlassCalibratorCalibrator
+    """
+    calibrator = IsotonicMulticlassCalibrator()
+    calibrator.fit(cal_uncs, cal_targets, cal_mask)
+    uncs = calibrator.apply(test_uncs)
+    torch.set_printoptions(precision=10)
```
Comment (shihchengli) [inline tests/unit/uncertainty/test_calibrators.py:187]: Is this necessary?

File: chemprop/uncertainty/calibrator.py (hunk 1 above)
Comment (KnathanM) [reply chemprop/uncertainty/calibrator.py:290]: Yeah, I don't think flake8 looks at the length of strings.

File: tests/unit/uncertainty/test_calibrators.py (hunk 3 above)
Comment (KnathanM) [reply tests/unit/uncertainty/test_calibrators.py:187]: Sorry that was from my testing and should have been removed.

File: chemprop/uncertainty/calibrator.py (hunk 1 above)
Comment (shihchengli) [reply chemprop/uncertainty/calibrator.py:292]: It seems that it is due to we ignore the E501 error in .flake8 file
https://github.com/chemprop/chemprop/blob/main/.flake8#L2

**Review by shihchengli — APPROVED** [review]

**Pull request merged**